src/main/java/it/gov/pagopa/payhub/activities/activity/debtposition/ValidateDebtPositionActivityImpl.java

```diff
@@ -0,0 +1,208 @@
+package it.gov.pagopa.payhub.activities.activity.debtposition;
+
+import it.gov.pagopa.payhub.activities.dao.AddressDao;
+import it.gov.pagopa.payhub.activities.dao.IngestionFlowFileDao;
+import it.gov.pagopa.payhub.activities.dto.IngestionFlowFileDTO;
+import it.gov.pagopa.payhub.activities.dto.PersonDTO;
+import it.gov.pagopa.payhub.activities.dto.PersonRequestDTO;
+import it.gov.pagopa.payhub.activities.dto.address.CityDTO;
+import it.gov.pagopa.payhub.activities.dto.address.NationDTO;
+import it.gov.pagopa.payhub.activities.dto.address.ProvinceDTO;
+import it.gov.pagopa.payhub.activities.dto.debtposition.*;
+import it.gov.pagopa.payhub.activities.exception.ValidationException;
+import it.gov.pagopa.payhub.activities.utility.Utilities;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Lazy
+@Service
+public class ValidateDebtPositionActivityImpl implements ValidateDebtPositionActivity {
+
+    private final IngestionFlowFileDao ingestionFlowFileDao;
+    private final AddressDao addressDao;
+
+    public ValidateDebtPositionActivityImpl(IngestionFlowFileDao ingestionFlowFileDao, AddressDao addressDao) {
+        this.ingestionFlowFileDao = ingestionFlowFileDao;
+        this.addressDao = addressDao;
+    }
+
+    public DebtPositionDTO validate(DebtPositionDTO debtPositionRequestDTO,
+                                    DebtPositionTypeOrgDTO debtPositionTypeOrgDTO) {
+        DebtPositionDTO debtPositionDTO = new DebtPositionDTO();
+
+        List<IngestionFlowFileDTO> flows = ingestionFlowFileDao.getIngestionFlowFilesByOrganization(
+                debtPositionTypeOrgDTO.getOrg().getOrgId(), true);
+
+        if (flows == null || flows.isEmpty())
+            throw new ValidationException("No flow was found for organization with id " + debtPositionTypeOrgDTO.getOrg().getOrgId());
+
+        debtPositionDTO.setIngestionFlowFile(flows.get(0));
+
+        if (debtPositionTypeOrgDTO.getDebtPositionType() == null ||
+                StringUtils.isBlank(debtPositionTypeOrgDTO.getDebtPositionType().getCode())) {
+            throw new ValidationException("Debt position type organization is mandatory");
+        }
+        debtPositionDTO.setDebtPositionTypeOrg(debtPositionTypeOrgDTO);
+
+        List<PaymentOptionDTO> paymentOptionsDTO = new ArrayList<>();
```

Comment:
again, you don't need to create a new list once you will reading/modify directly the input debtPosition